README.md
script.globalsearch
===================


changelog.txt
v9.0.4
- add functionality to navigate back
- add support for searching musicvideos by artist name
- add support for originaltitle search
- add support for tv show actors
- add videoresolution flag
- small skin layout changes

v9.0.3
- fix listing of pvr results

v9.0.2
- cosmetics

v9.0.1
- bump

v9.0.0
- changes for python 3

v8.0.12
- fix sorting of album songs and season episodes

v8.0.11
- leia bump

v8.0.10
- implement play/resume option

v8.0.9
- improve support for compilation albums

v8.0.8
- fix fanart for seasons and episodes

v8.0.7
- language

v8.0.6
- move repo to gitlab

v8.0.5
- favourites fix

v8.0.4
- add to favourites context menu item
- case insensitive search for actors and directors

v9.0.0
- changes for python 3

v8.0.3
- reset menu on init

v8.0.2
- fix content selection when using a horizontal menu
- fixed incorrect listitem.title for episodes
- show result counts in the menu
- fix content selection when using a horizontal menu

v8.0.1
- fix passing search categories to the script
- save the selected viewmode
- close globalsearch if no is selected when promted for a new search

v8.0.0
- rewrite for Kodi Leia

v7.3.1
- remove test settings

v7.3.0
- fix epg section
- removed channelnumber infolabel for epg items
- removed path infolabel for epg items

v7.2.4
- add originaltitle support

v7.2.3
- cosmetics radiobuttons
- don't show empty contextmenu

v7.2.2
- cleanup

v7.2.1
- skin cosmetics

v7.2.0
- use context menu from kodi

v7.1.0
- add direcor search support

v7.0.5
- code refactor
- add missing skin support for livetv
- fix infodialog buttons

v7.0.4
- ask user to search again when no results are found

v7.0.3
- new skin
- remove ok button in info dialog

v7.0.2
- return empty userrating instead of 0

v7.0.1
- remove starrating
- add support for userrating
- epg runtime is an int now

v7.0.0
- remove simplejson support

v6.0.2
- language update
- cosmetics

v6.0.1
- update action id's for jarvis

v5.0.0
- language update

v4.0.3
- fix README

v4.0.2
- rating fixes

v4.0.1
- fix channel id

v4.0.0
- add support for epg search
- fix fanart in infodialog

v3.0.4
- update language files from Transifex

v3.0.3
- skin changes for helix

v3.0.2
- fix crash on exit

v3.0.1
- fix timing issue

v3.0.0
- kodi name change

v2.1.1
- fix crash on skins that don't support actor search

v2.1.0
- add actor search option

v2.0.8
- add xml header

v2.0.7
- language update

v2.0.6
- language update

v2.0.5
- gotham release

v2.0.4
- more path fixes for gotham

v2.0.3
- db path changes for gotham

v2.0.2
- removed search by originaltitle as it's not supported in json

v2.0.1
- properly round runtime value

v2.0.0
- re-code the context menu

v1.1.1
- display runtime in minutes

v1.1.0
- added support for tv show banner and poster

v1.0.6
- fix runtime

v1.0.5
- fix encoding issues

v1.0.4
- rename fanart property

v1.0.3
- update code to latest json-rpc changes

v1.0.2
- use json-rpc to play albums

v1.0.1
- use action.getId() instead of action

v1.0.0
- adjust code for frodo

v0.2.7
- ignore empty searchstrings

v0.2.6
- added hungarian language file

v0.2.5
- also search originaltitle for movies

v0.2.4
- add searchstring window property
- allow dialogfullscreeninfo to be used

v0.2.3
- use label for seasons
- sort by label instead of title

v0.2.2
- use title instead of label

v0.2.1
- fix unicode in path issue

v0.2.0
- add seperate context menu items for find seasons and find episodes
- add seperate context menu items for find albums and find songs

v0.1.9
- add starrating

v0.1.8
- remove unused code

v0.1.7
- ignore possible encoding errors
- improve videoresolution matching

v0.1.6
- check if streamdetails items exist before calling them
- show artistname instead of recordlabel for musicvideos in gui

v0.1.5
- use simplejson instead of regex

v0.1.4
- clean plot and description

v0.1.3
- added option for skins to specify search categories

v0.1.2
- removed some debug logging
- added Portuguese (Brazil) translation
- added Italian translation

v0.1.1
- adjusted to latest json-rpc changes
- fixed fetching of streamdetails

v0.1.0
- initial release

v0.0.6 (beta 6)
- fixed json query when compactoutput was set to false

v0.0.5 (beta 5)
- fixed hidden item showed up on info dialog

v0.0.4 (beta 4)
- speed up list creation
- fixed issue with + in searchstring
- fixed issue with brackets in searchstring
- added key actions to close the context and info dialog
- added writer info for movies
- sort items by title
- simplified code for searching episodes

v0.0.3 (beta 3)
- added mediaflags

v0.0.2 (beta 2)
- use less strict regex matching
- don't use a window property to pass the searchstring
- added 'find album' option to songs context menu
- remove unneeded regex match in play_album code
- use built-in python method to clear the playlist
- simplified code to pass listitem to infodialog
- made infodialog fully skinnable

v0.0.1 (beta 1)
- initial beta


default.py
import os
import sys
from urllib.parse import unquote_plus
import xbmc
import xbmcaddon

ADDON = xbmcaddon.Addon()
LANGUAGE = ADDON.getLocalizedString
CWD = ADDON.getAddonInfo('path')

if (__name__ == '__main__'):
    try:
        params = dict(arg.split('=') for arg in sys.argv[ 1 ].split('&'))
    except:
        params = {}
    searchstring = params.get('searchstring','')
    searchstring = unquote_plus(searchstring)
    if searchstring == '':
        keyboard = xbmc.Keyboard('', LANGUAGE(32101), False)
        keyboard.doModal()
        if (keyboard.isConfirmed()):
            searchstring = keyboard.getText()
    else:
        del params['searchstring']
    if searchstring:
        from lib import gui
        ui = gui.GUI('script-globalsearch.xml', CWD, 'default', '1080i', True, searchstring=searchstring, params=params)
        ui.doModal()
        del ui


resources/lib/utils.py
import sys
import re
import xbmc
import xbmcgui

ADDON = sys.modules['__main__'].ADDON
ADDONID = sys.modules['__main__'].ADDONID
ADDONVERSION = sys.modules['__main__'].ADDONVERSION
LANGUAGE = sys.modules['__main__'].LANGUAGE
CWD = sys.modules['__main__'].CWD

ACTION_CANCEL_DIALOG = (9, 10, 92, 216, 247, 257, 275, 61467, 61448,)
ACTION_CONTEXT_MENU = (117,)
ACTION_OSD = (107, 163,)
ACTION_SHOW_GUI = (18,)
ACTION_SHOW_INFO = (11,)

MOVIELABELS = ["genre", "country", "year", "top250", "setid", "rating", "userrating", "playcount", "cast", "director", "mpaa", "plot", "plotoutline", "title", "originaltitle", "sorttitle", 
               "runtime", "studio", "tagline", "writer", "premiered", "set", "imdbnumber", "lastplayed", "votes", "trailer", "dateadded", "streamdetails", "art"]

TVSHOWLABELS = ["genre", "year", "episode", "season", "rating", "userrating", "playcount", "cast", "mpaa", "plot", "title", "originaltitle", "sorttitle", "runtime", "studio", "premiered", 
                "imdbnumber", "lastplayed", "votes", "dateadded", "art"]

SEASONLABELS = ["episode", "season", "showtitle", "tvshowid", "userrating", "watchedepisodes", "playcount", "art"]

EPISODELABELS = ["episode", "season", "rating", "userrating", "playcount", "cast", "director", "plot", "title", "originaltitle", "runtime", "writer", "showtitle", "firstaired", "lastplayed", 
                 "votes", "dateadded", "streamdetails", "art"]

MUSICVIDEOLABELS = ["genre", "year", "rating", "userrating", "playcount", "director", "plot", "title", "runtime", "studio", "premiered", "lastplayed", "album", "artist", "dateadded", 
                    "streamdetails", "art"]

ARTISTLABELS = ["genre", "description", "formed", "disbanded", "born", "yearsactive", "died", "mood", "style", "instrument", "thumbnail", "fanart"]

ALBUMLABELS = ["title", "description", "albumlabel", "artist", "genre", "year", "thumbnail", "fanart", "theme", "type", "mood", "style", "rating", "userrating"]

SONGLABELS = ["title", "artist", "album", "genre", "duration", "year", "file", "thumbnail", "fanart", "comment", "rating", "userrating", "track", "playcount", "albumid"]

CATEGORIES = {
              'movies':{
                        'order':1, 
                        'enabled':False, 
                        'type':'movies', 
                        'method':'VideoLibrary.GetMovies', 
                        'properties':MOVIELABELS, 
                        'sort':'title', 
                        'rule':'{"field":"title", "operator":"contains", "value":"%s"}',
                        'cast':True, 
                        'streamdetails':True, 
                        'label':342, 
                        'icon':'DefaultVideo.png', 
                        'media':'video', 
                        'control':110
                       }, 
              'tvshows':{
                         'order':2, 
                         'enabled':False, 
                         'type':'tvshows', 
                         'method':'VideoLibrary.GetTVShows', 
                         'properties':TVSHOWLABELS, 
                         'sort':'label', 
                         'rule':'{"field":"title", "operator":"contains", "value":"%s"}',
                         'cast':True, 
                         'streamdetails':False, 
                         'label':20343, 
                         'icon':'DefaultVideo.png', 
                         'media':'video', 
                         'control':120
                        }, 
              'episodes':{
                          'order':3, 
                          'enabled':False, 
                          'type':'episodes', 
                          'method':'VideoLibrary.GetEpisodes', 
                          'properties':EPISODELABELS, 
                          'sort':'title', 
                          'rule':'{"field":"title", "operator":"contains", "value":"%s"}',
                          'cast':True, 
                          'streamdetails':True, 
                          'label':20360, 
                          'icon':'DefaultVideo.png', 
                          'media':'video', 
                          'control':140
                         }, 
              'musicvideos':{
                             'order':4, 
                             'enabled':False, 
                             'type':'musicvideos', 
                             'method':'VideoLibrary.GetMusicVideos', 
                             'properties':MUSICVIDEOLABELS, 
                             'sort':'label', 
                             'rule':'{"field":"title", "operator":"contains", "value":"%s"}',
                             'cast':False, 
                             'streamdetails':True, 
                             'label':20389, 
                             'icon':'DefaultVideo.png', 
                             'media':'video', 
                             'control':150
                            }, 
              'artists':{
                         'order':5, 
                         'enabled':False, 
                         'type':'artists', 
                         'method':'AudioLibrary.GetArtists', 
                         'properties':ARTISTLABELS, 
                         'sort':'label', 
                         'rule':'{"field": "artist", "operator": "contains", "value": "%s"}',
                         'cast':False, 
                         'streamdetails':False, 
                         'label':133, 
                         'icon':'DefaultArtist.png', 
                         'media':'music', 
                         'control':160
                        }, 
              'albums':{
                        'order':6, 
                        'enabled':False, 
                        'type':'albums', 
                        'method':'AudioLibrary.GetAlbums', 
                        'properties':ALBUMLABELS, 
                        'sort':'label', 
                        'rule':'{"field": "album", "operator": "contains", "value": "%s"}',
                        'cast':False, 
                        'streamdetails':False, 
                        'label':132, 
                        'icon':'DefaultAlbumCover.png', 
                        'media':'music', 
                        'control':170
                       }, 
              'songs':{
                       'order':7, 
                       'enabled':False, 
                       'type':'songs', 
                       'method':'AudioLibrary.GetSongs', 
                       'properties':SONGLABELS, 
                       'sort':'title', 
                       'rule':'{"field": "title", "operator": "contains", "value": "%s"}',
                       'cast':False, 
                       'streamdetails':False, 
                       'label':134, 
                       'icon':'DefaultAudio.png', 
                       'media':'music', 
                       'control':180
                      }, 
              'epg':{
                     'order':9, 
                     'enabled':False, 
                     'type':'epg'
                    },  
              'actors':{
                        'order':10, 
                        'enabled':False, 
                        'type':'movies', 
                        'method':'VideoLibrary.GetMovies', 
                        'properties':MOVIELABELS, 
                        'sort':'title', 
                        'rule':'{"field":"actor", "operator":"contains", "value":"%s"}',
                        'cast':True, 
                        'streamdetails':True, 
                        'label':344, 
                        'icon':'DefaultVideo.png', 
                        'media':'video', 
                        'control':200
                       },
              'directors':{
                           'order':11, 
                           'enabled':False, 
                           'type':'movies', 
                           'method':'VideoLibrary.GetMovies', 
                           'properties':MOVIELABELS, 
                           'sort':'title', 
                           'rule':'{"field":"director", "operator":"contains", "value":"%s"}',
                           'cast':True, 
                           'streamdetails':True, 
                           'label':20348, 
                           'icon':'DefaultVideo.png', 
                           'media':'video', 
                           'control':210
                          }, 
              'tvshowseasons':{
                               'order':11, 
                               'enabled':False, 
                               'type':'seasons', 
                               'method':'VideoLibrary.GetSeasons', 
                               'properties':SEASONLABELS, 
                               'sort':'label', 
                               'rule':'{"tvshowid":%s}',
                               'cast':False, 
                               'streamdetails':False, 
                               'label':20373, 
                               'icon':'DefaultVideo.png', 
                               'media':'video', 
                               'control':130
                              },
              'tvshowepisodes':{
                                'order':12, 
                                'enabled':False, 
                                'type':'episodes', 
                                'method':'VideoLibrary.GetEpisodes', 
                                'properties':EPISODELABELS, 
                                'sort':'title', 
                                'rule':'{"tvshowid":%s}',
                                'cast':True, 
                                'streamdetails':True, 
                                'label':20360, 
                                'icon':'DefaultVideo.png', 
                                'media':'video', 
                                'control':140
                               },
              'artistalbums':{
                              'order':13, 
                              'enabled':False, 
                              'type':'albums', 
                              'method':'AudioLibrary.GetAlbums', 
                              'properties':ALBUMLABELS, 
                              'sort':'label', 
                              'rule':'{"artistid": %s}',
                              'cast':False, 
                              'streamdetails':False, 
                              'label':132, 
                              'icon':'DefaultAlbumCover.png', 
                              'media':'music', 
                              'control':170
                             },
              'artistsongs':{
                             'order':14, 
                             'enabled':False, 
                             'type':'songs', 
                             'method':'AudioLibrary.GetSongs', 
                             'properties':SONGLABELS, 
                             'sort':'title', 
                             'rule':'{"artistid": %s}',
                             'cast':False, 
                             'streamdetails':False, 
                             'label':134, 
                             'icon':'DefaultAudio.png', 
                             'media':'music', 
                             'control':180
                            }, 
              'songalbum':{
                           'order':15, 
                           'enabled':False, 
                           'type':'albums', 
                           'method':'AudioLibrary.GetAlbums', 
                           'properties':ALBUMLABELS, 
                           'sort':'label', 
                           'rule':'{"albumid": %s}',
                           'cast':False, 
                           'streamdetails':False, 
                           'label':132, 
                           'icon':'DefaultAlbumCover.png', 
                           'media':'music', 
                           'control':170
                          }
             }

def log(txt):
    if isinstance(txt,str):
        txt = txt.decode('utf-8')
    message = u'%s: %s' % (ADDONID, txt)
    xbmc.log(msg=message.encode('utf-8'), level=xbmc.LOGDEBUG)


lib/defs.py
import sys
import re
import xbmc
import xbmcgui
import xbmcaddon

ADDON = xbmcaddon.Addon()
ADDONID = ADDON.getAddonInfo('id')
ADDONVERSION = ADDON.getAddonInfo('version')
LANGUAGE = ADDON.getLocalizedString

ACTION_CANCEL_DIALOG = (9, 10, 92, 216, 247, 257, 275, 61467, 61448,)
ACTION_CONTEXT_MENU = (117,)
ACTION_SHOW_INFO = (11,)

SEARCHBUTTON = 990
SEARCHCATEGORY = 991
NORESULTS = 999
MENU = 9000

MOVIELABELS = ["genre", "country", "year", "top250", "setid", "rating", "userrating", "playcount", "director", "mpaa", "plot", "plotoutline", "title", "originaltitle", "sorttitle",
               "runtime", "studio", "tagline", "writer", "premiered", "set", "imdbnumber", "lastplayed", "votes", "trailer", "dateadded", "streamdetails", "art", "file", "resume"]

TVSHOWLABELS = ["genre", "year", "episode", "season", "rating", "userrating", "playcount", "mpaa", "plot", "title", "originaltitle", "sorttitle", "runtime", "studio", "premiered",
                "imdbnumber", "lastplayed", "votes", "dateadded", "art", "watchedepisodes", "file"]

SEASONLABELS = ["episode", "season", "showtitle", "tvshowid", "userrating", "watchedepisodes", "playcount", "art"]

EPISODELABELS = ["episode", "season", "rating", "userrating", "playcount", "director", "plot", "title", "originaltitle", "runtime", "writer", "showtitle", "firstaired", "lastplayed",
                 "votes", "dateadded", "streamdetails", "art", "file", "resume"]

MUSICVIDEOLABELS = ["genre", "year", "rating", "userrating", "playcount", "director", "plot", "title", "runtime", "studio", "premiered", "lastplayed", "album", "artist", "dateadded",
                    "streamdetails", "art", "file", "resume"]

ARTISTLABELS = ["genre", "description", "formed", "disbanded", "born", "yearsactive", "died", "mood", "style", "instrument", "thumbnail", "fanart", "art"]

ALBUMLABELS = ["title", "description", "albumlabel", "artist", "genre", "year", "thumbnail", "fanart", "art", "theme", "type", "mood", "style", "rating", "userrating", "artistid"]

SONGLABELS = ["title", "artist", "album", "genre", "duration", "year", "file", "thumbnail", "fanart", "comment", "art", "rating", "userrating", "track", "playcount", "artistid", "albumid"]

ACTORLABELS = ["cast"]

DIRECTORLABELS = ["director"]

CATEGORIES = {
              'movies':{
                        'order':1,
                        'enabled':False,
                        'type':'movies',
                        'content':'movies',
                        'method':'VideoLibrary.GetMovies',
                        'properties':MOVIELABELS,
                        'sort':'title',
                        'rule':'"filter":{{"or":[{{"field":"title", "operator":"contains", "value":"{query}"}}, {{"field":"originaltitle", "operator":"contains", "value":"{query}"}}]}}',
                        'streamdetails':True,
                        'label':342,
                        'icon':'DefaultVideo.png',
                        'menuthumb':'globalsearch-icon-movies.png',
                        'media':'video'
                       },
              'tvshows':{
                         'order':2,
                         'enabled':False,
                         'type':'tvshows',
                         'content':'tvshows',
                         'method':'VideoLibrary.GetTVShows',
                         'properties':TVSHOWLABELS,
                         'sort':'label',
                         'rule':'"filter":{{"field":"title", "operator":"contains", "value":"{query}"}}',
                         'streamdetails':False,
                         'label':20343,
                         'icon':'DefaultVideo.png',
                         'menuthumb':'globalsearch-icon-tvshows.png',
                         'media':'video'
                        },
              'episodes':{
                          'order':3,
                          'enabled':False,
                          'type':'episodes',
                          'content':'episodes',
                          'method':'VideoLibrary.GetEpisodes',
                          'properties':EPISODELABELS,
                          'sort':'title',
                          'rule':'"filter":{{"field":"title", "operator":"contains", "value":"{query}"}}',
                          'streamdetails':True,
                          'label':20360,
                          'icon':'DefaultVideo.png',
                          'menuthumb':'globalsearch-icon-episodes.png',
                          'media':'video'
                         },
              'musicvideos':{
                             'order':4,
                             'enabled':False,
                             'type':'musicvideos',
                             'content':'musicvideos',
                             'method':'VideoLibrary.GetMusicVideos',
                             'properties':MUSICVIDEOLABELS,
                             'sort':'label',
                             'rule':'"filter":{{"field":"title", "operator":"contains", "value":"{query}"}}',
                             'streamdetails':True,
                             'label':20389,
                             'icon':'DefaultVideo.png',
                             'menuthumb':'globalsearch-icon-musicvideos.png',
                             'media':'video'
                            },
              'artists':{
                         'order':5,
                         'enabled':False,
                         'type':'artists',
                         'content':'artists',
                         'method':'AudioLibrary.GetArtists',
                         'properties':ARTISTLABELS,
                         'sort':'label',
                         'rule':'"filter":{{"field": "artist", "operator": "contains", "value": "{query}"}}',
                         'streamdetails':False,
                         'label':133,
                         'icon':'DefaultArtist.png',
                         'menuthumb':'globalsearch-icon-artists.png',
                         'media':'music'
                        },
              'albums':{
                        'order':6,
                        'enabled':False,
                        'type':'albums',
                        'content':'albums',
                        'method':'AudioLibrary.GetAlbums',
                        'properties':ALBUMLABELS,
                        'sort':'label',
                        'rule':'"filter":{{"field": "album", "operator": "contains", "value": "{query}"}}',
                        'streamdetails':False,
                        'label':132,
                        'icon':'DefaultAlbumCover.png',
                        'menuthumb':'globalsearch-icon-albums.png',
                        'media':'music'
                       },
              'songs':{
                       'order':7,
                       'enabled':False,
                       'type':'songs',
                       'content':'songs',
                       'method':'AudioLibrary.GetSongs',
                       'properties':SONGLABELS,
                       'sort':'title',
                       'rule':'"filter":{{"field": "title", "operator": "contains", "value": "{query}"}}',
                       'streamdetails':False,
                       'label':134,
                       'icon':'DefaultAudio.png',
                       'menuthumb':'globalsearch-icon-songs.png',
                       'media':'music'
                      },
              'livetv':{
                     'order':9,
                     'enabled':False,
                     'type':'livetv',
                     'content':'livetv',
                     'label':19069,
                     'menuthumb':'globalsearch-icon-livetv.png'
                    },
              'actors':{
                        'order':10,
                        'enabled':False,
                        'type':'actors',
                        'content':'movies',
                        'method':'VideoLibrary.GetMovies',
                        'properties':ACTORLABELS,
                        'sort':'title',
                        'rule':'"filter":{{"field":"actor", "operator":"contains", "value":"{query}"}}',
                        'streamdetails':False,
                        'label':32221,
                        'icon':'DefaultActor.png',
                        'menuthumb':'globalsearch-icon-actors.png',
                        'media':''
                       },
              'directors':{
                           'order':11,
                           'enabled':False,
                           'type':'directors',
                           'content':'movies',
                           'method':'VideoLibrary.GetMovies',
                           'properties':DIRECTORLABELS,
                           'sort':'title',
                           'rule':'"filter":{{"field":"director", "operator":"contains", "value":"{query}"}}',
                           'streamdetails':False,
                           'label':20348,
                           'icon':'DefaultDirector.png',
                           'menuthumb':'globalsearch-icon-directors.png',
                           'media':''
                          },
              'tvactors':{
                          'order':12,
                          'enabled':False,
                          'type':'tvactors',
                          'content':'tvshows',
                          'method':'VideoLibrary.GetTVShows',
                          'properties':ACTORLABELS,
                          'sort':'title',
                          'rule':'"filter":{{"field":"actor", "operator":"contains", "value":"{query}"}}',
                          'streamdetails':False,
                          'label':32222,
                          'icon':'DefaultActor.png',
                          'menuthumb':'globalsearch-icon-actors.png',
                          'media':''
                         },
              'actormovies':{
                        'order':13,
                        'enabled':False,
                        'type':'movies',
                        'content':'movies',
                        'method':'VideoLibrary.GetMovies',
                        'properties':MOVIELABELS,
                        'sort':'title',
                        'rule':'"filter":{{"field":"actor", "operator":"contains", "value":"{query}"}}',
                        'streamdetails':True,
                        'label':342,
                        'icon':'DefaultVideo.png',
                        'menuthumb':'globalsearch-icon-movies.png',
                        'media':'video'
                       },
              'directormovies':{
                        'order':14,
                        'enabled':False,
                        'type':'movies',
                        'content':'movies',
                        'method':'VideoLibrary.GetMovies',
                        'properties':MOVIELABELS,
                        'sort':'title',
                        'rule':'"filter":{{"field":"director", "operator":"contains", "value":"{query}"}}',
                        'streamdetails':True,
                        'label':342,
                        'icon':'DefaultVideo.png',
                        'menuthumb':'globalsearch-icon-movies.png',
                        'media':'video'
                       },
              'actortvshows':{
                              'order':15,
                              'enabled':False,
                              'type':'tvshows',
                              'content':'tvshows',
                              'method':'VideoLibrary.GetTVShows',
                              'properties':TVSHOWLABELS,
                              'sort':'title',
                              'rule':'"filter":{{"field":"actor", "operator":"contains", "value":"{query}"}}',
                              'streamdetails':False,
                              'label':20343,
                              'icon':'DefaultVideo.png',
                              'menuthumb':'globalsearch-icon-tvshows.png',
                              'media':'video'
                             },
              'tvshowseasons':{
                               'order':16,
                               'enabled':False,
                               'type':'tvshowseasons',
                               'content':'seasons',
                               'method':'VideoLibrary.GetSeasons',
                               'properties':SEASONLABELS,
                               'sort':'label',
                               'rule':'"tvshowid":{query}',
                               'streamdetails':False,
                               'label':20373,
                               'icon':'DefaultVideo.png',
                               'menuthumb':'globalsearch-icon-seasons.png',
                               'media':'video'
                              },
              'seasonepisodes':{
                                'order':17,
                                'enabled':False,
                                'type':'seasonepisodes',
                                'content':'episodes',
                                'method':'VideoLibrary.GetEpisodes',
                                'properties':EPISODELABELS,
                                'sort':'episode',
                                'rule':'"tvshowid":{query0}, "season":{query1}',
                                'streamdetails':True,
                                'label':20360,
                                'icon':'DefaultVideo.png',
                                'menuthumb':'globalsearch-icon-episodes.png',
                                'media':'video'
                               },
              'artistalbums':{
                              'order':18,
                              'enabled':False,
                              'type':'artistalbums',
                              'content':'albums',
                              'method':'AudioLibrary.GetAlbums',
                              'properties':ALBUMLABELS,
                              'sort':'label',
                              'rule':'"filter":{{"artistid":{query}}}',
                              'streamdetails':False,
                              'label':132,
                              'icon':'DefaultAlbumCover.png',
                              'menuthumb':'globalsearch-icon-albums.png',
                              'media':'music'
                             },
              'albumsongs':{
                             'order':19,
                             'enabled':False,
                             'type':'albumsongs',
                             'content':'songs',
                             'method':'AudioLibrary.GetSongs',
                             'properties':SONGLABELS,
                             'sort':'track',
                             'rule':'"filter":{{"albumid":{query}}}',
                             'streamdetails':False,
                             'label':134,
                             'icon':'DefaultAudio.png',
                             'menuthumb':'globalsearch-icon-songs.png',
                             'media':'music'
                            }
             }
